test/e2e/framework/framework.go

```diff
@@ -884,3 +884,18 @@ func GetLogToFileFunc(file *os.File) func(format string, args ...interface{}) {
 		writer.Flush()
 	}
 }
+
+const (
+	// preconfiguredRuntimeHandler is the name of the runtime handler that is expected to be
+	// preconfigured in the test environment.
+	preconfiguredRuntimeHandler = "test-handler"
+)
+
+// PreconfiguredRuntimeClassHandler returns configured runtime handler.
+func PreconfiguredRuntimeClassHandler() string {
```

Comment:
i think we need a better location for this function instead of `framework.go`.
possibly `/test/e2e/framework/node/runtime.go`, @oomichi WDYT?


/hold

Reply:
Either of `test/e2e/framework/{node,pod}/runtime.go` works for me.

Reply:
it would also be really nice if we documented this somewhere, other than kube-up, clusters don't tend to have "test-handler" registered, so they're only passing whenw e run docker and ~ skip them ... (?)

Reply:
I understand _why_ it's like this, but it's a bit annoying to track this down and modify cluster deployments to support the test. I imagine most just don't have this setup and aren't aware of it.

Reply:
> I understand _why_ it's like this, but it's a bit annoying to track this down and modify cluster deployments to support the test. I imagine most just don't have this setup and aren't aware of it.

Moving it to the `test/e2e/framework/node/runtime.go` would make the e2e node package depends on the framework. I checked out several other e2e packages which do not have this dependency.

Reply:
I mean documenting somewhere that the cluster must have this magical `test-handler` configured

Reply:
Makes sense. Since this PR is just moving the code that's already there & needs to get in to 1.16, can we follow up on that?

I think the better approach would be to make it a variable that get's filled in by the test setup, rather than branching on the runtime, just make the handler parameter explicit (and skip the test if it's missing). But again, I'd like to get this in to 1.16 as is, and follow up with that change.

Reply:
SGTM 👍

Reply:
I created an issue to track this https://github.com/kubernetes/kubernetes/issues/82213